### PR TITLE
[SPARK-25663][SPARK-25661][SQL][TEST] Refactor BuiltInDataSourceWriteBenchmark, DataSourceWriteBenchmark and AvroWriteBenchmark to use main method

### PR DESCRIPTION
**Pull request by yucai**
## What changes were proposed in this pull request?

Refactor BuiltInDataSourceWriteBenchmark, DataSourceWriteBenchmark and AvroWriteBenchmark to use main method.

```
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.benchmark.BuiltInDataSourceWriteBenchmark"

SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "avro/test:runMain org.apache.spark.sql.execution.benchmark.AvroWriteBenchmark"
```
## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by yucai** [comment]
@gengliangwang @wangyum @dongjoon-hyun help review.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
@yucai . We (@yucai and @gengliangwang and me) know what is this for, but please don't piggy-back (hide) something like this. This kind of change requires another explicit PR in general.

**Comment by yucai** [comment]
@dongjoon-hyun Originally, I want to do two things in this PR.
1.  Make `mainArgs` correctly set in `BenchmarkBase`.
2. Include an example to use `mainArgs`: refactor `DataSourceWriteBenchmark` and `BuiltInDataSourceWriteBenchmark` to use main method.

But, refactor `DataSourceWriteBenchmark` will lead to refactor `AvroWriteBenchmark`.

Any suggestion how to split PR?

**Comment by dongjoon-hyun** [comment]
At least, the following is worth for a separate PR because it's orthogonal `Refactor ... to use main method`.
```
1. Make mainArgs correctly set in BenchmarkBase.
```

One PR had better have one theme. Putting different themes into one PR together is not a good practice.

Please start with the minimal one. If the committer asks some example, then add that later. That's better.

**Comment by gengliangwang** [comment]
Personally I am against accessing the main args in such way. It looks a bit ugly. 
But if we have to move everything to `BenchmarkBase`, then maybe this is the way.

**Comment by yucai** [comment]
Current implementation misses main args, but some suite would need it anyway. 
Let's can discuss #22872.

**Comment by yucai** [comment]
@dongjoon-hyun I used #22872 to make main args accessible for `BenchmarkBase`'s subclass, this PR is mainly for refactoring `DataSourceWriteBenchmark` and `BuiltInDataSourceWriteBenchmark`. But it will lead to `AvroWriteBenchmark` refactor, so I put them in one PR, kindly let me know if you have better idea.

File: external/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroWriteBenchmark.scala [hunk 1]
```diff
@@ -19,22 +19,17 @@ package org.apache.spark.sql.execution.benchmark
 
 /**
  * Benchmark to measure Avro data sources write performance.
- * Usage:
- * 1. with spark-submit: bin/spark-submit --class <this class> <spark sql test jar>
- * 2. with sbt: build/sbt "avro/test:runMain <this class>"
+ * To run this benchmark:
+ * {{{
+ *   1. without sbt: bin/spark-submit --class <this class>
```
Comment (wangyum) [inline external/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroWriteBenchmark.scala]: Please add `avro`:
```
bin/spark-submit --class <this class> --jars ./core/target/spark-core_2.11-3.0.0-SNAPSHOT-<spark core test jar>,<spark catalyst test jar>,<spark sql test jar> <spark avro test jar>
```

Comment (yucai) [reply external/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroWriteBenchmark.scala]: @yucai Good catch! I think it needs `<spark avro jar>`, added.

**Comment by yucai** [comment]
@dongjoon-hyun Tests have been passed.

**Review by dongjoon-hyun — COMMENTED** [review]
@yucai . Almost looks good. Please update the PR description according to the PR content and merge https://github.com/yucai/spark/pull/6 .
- Fix
```
- SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.benchmark.AvroWriteBenchmark"
+ SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "avro/test:runMain org.apache.spark.sql.execution.benchmark.AvroWriteBenchmark"
```
- Remove
```
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.benchmark.BuiltInDataSourceWriteBenchmark Parquet ORC"
```

**Comment by dongjoon-hyun** [comment]
Thank you, @yucai , @gengliangwang  and @wangyum .
Merged to master.

**Pull request closed without merging**